lib/Conversion/HandshakeToHW/HandshakeToHW.cpp

```diff
@@ -650,6 +650,7 @@ class HandshakeConversionPattern : public OpConversionPattern<T> {
     for (auto &input : inputs)
       valids.push_back(input.valid);
     Value allValid = s.bAnd(valids);
+    output.valid->setValue(allValid);
```

Comment:
This change might require a removal of a similar line in `buildUnitRateLogic`: https://github.com/llvm/circt/blob/main/lib/Conversion/HandshakeToHW/HandshakeToHW.cpp#L727